pkg/prove/proof.go

```diff
@@ -107,6 +117,49 @@ func txSharePosition(txs types.Txs, txIndex uint64) (startSharePos, endSharePos
 	return startSharePos, endSharePos, nil
 }
 
+// MsgSharesPosition returns the start and end positions for the shares
+// where a given message, referenced by its wrapped pfb transaction, was published at.
+// Note: only supports transactions containing a single message
+func MsgSharesPosition(tx types.Tx) (beginShare uint64, endShare uint64, err error) {
+	unwrappedTx, isMalleated := types.UnmarshalIndexWrapper(tx)
+	if !isMalleated {
+		return beginShare, endShare, fmt.Errorf("not a malleated tx")
+	}
+
+	encCfg := encoding.MakeConfig(blobmodule.AppModuleBasic{})
+	decoder := encCfg.TxConfig.TxDecoder()
+
+	decodedTx, err := decoder(unwrappedTx.Tx)
+	if err != nil {
+		return beginShare, endShare, err
+	}
+
+	if len(decodedTx.GetMsgs()) == 0 {
```

Comment:
[note to reviewers]
This PR only proves one message per blob. We add multiple messages later

Reply:
https://github.com/celestiaorg/celestia-app/issues/1234

Reply:
[blocking question]

> This helper method only proves one message per blob. We add multiple messages later

should we check for or something similar then?
```suggestion
	if len(decodedTx.GetMsgs()) != 1 {
```

note, preview does not also change the error message below which we'll also have to change should this be accepted

Reply:
There is a subsequent check:
```
	if len(decodedTx.GetMsgs()) > 1 {
		return beginShare, endShare, fmt.Errorf("PayForBlob contains multiple messages and this is not currently supported")
	}
```
that should catch that

Reply:
There is a subsequent check:
```
	if len(decodedTx.GetMsgs()) > 1 {
		return beginShare, endShare, fmt.Errorf("PayForBlob contains multiple messages and this is not currently supported")
	}
```
that should catch that